frontend-react/src/components/Admin/AdminFormEdit.tsx

```diff
@@ -86,10 +87,20 @@ export const TextAreaComponent = (params: {
                     onBlur={(e) => {
                         const text =
                             e?.target?.value || (defaultnullvalue as string);
-                        const result = checkTextAreaJson(text, key, inputRef);
-                        if (result !== false) {
-                            // checkTextAreaJson made sure the following call won't throw.
-                            params.savefunc(result);
+                        const { valid, offset, errorMsg } = checkJson(text);
+                        if (valid) {
+                            // checkJson made sure the following JSON.parse won't throw.
+                            params.savefunc(JSON.parse(text));
+                        } else {
+                            showError(
+                                `JSon data generated an error "${errorMsg}"`
+                            );
+
+                            // now select the problem area inside the TextArea
+                            const start = Math.max(offset - 4, 0); // don't let go negative
+                            const end = Math.min(offset + 4, text.length); // don't let go past len
+                            inputRef?.current?.focus();
+                            inputRef?.current?.setSelectionRange(start, end);
                         }
```

Comment:
The above section was moved from misc. It needs deep access to the inputRef, so it makes more sense to have it in the component itself. I did this because other location needed a generic `checkJson()` feature without specifics.